### DISCUSSION TITLE
Dirichlet boundary condition for custom grid config

### DISCUSSION THREAD

**Discussion by kallegrens** [zwicker-group/py-pde#334] [Q&A]
Hello!

I'm a beginner using py-pde. I'm wondering how I can define custom Dirichlet conditions for a grid. Is it possible to pass a numpy array with my config? See the image below for clarity. 

Thanks in advance!
<img width="355" alt="Screenshot 2022-11-19 at 20 04 00" src="https://user-images.githubusercontent.com/105279176/202867550-db26bf40-6905-4c34-b31a-964c6cf375a3.png">

**Comment by david-zwicker** [comment]
I'm not quite sure what you're asking and I don't know how to interpret the graphics. If you're asking whether `py-pde` supports general geometries and boundaries of arbitrary shape, then I'm afraid I need to disappoint you.

**Reply by kallegrens** [reply]
When solving the Laplacian I want to be able to set spatial boundary conditions individually for cells on my grid. The graphics correspond to the electric potential for some configuration with the goal to simulate arcs of lightning. I would need to solve the Laplacian iteratively and pass a new (slightly modified) array of Dirichlet boundary conditions with every iteration.

**Reply by david-zwicker** [reply]
If you are talking about spatially dependent boundary conditions, then this is possible. You can simply supply an expression or a numpy array as a boundary condition, e.g., `bc = {"value": np.arange(32)}`.

**Reply by kallegrens** [reply]
Thank you!  I will try this!
